engine/tests/mechanics/games/test_blackjack_game.py

```
@@ -172,3 +173,72 @@ def test_context_exports_blackjack_pressure_signals(self) -> None:
+class TestCardsAsTokens:
+    """Cards are tokens over a shared deck definition."""
+
+    def test_a_card_is_a_canonical_token(self) -> None:
+        assert issubclass(PlayingCard, Token)
+
+    def test_rank_and_suit_come_from_the_frozen_definition(self) -> None:
+        card = PlayingCard.of(12, "d")
+
+        assert card.short_name == "QD"
+        assert (card.rank, card.suit) == (12, "d")
+        assert card.token_from == "QD"
```

Ruff (0.16.3)

[error] Possible hardcoded password assigned to: "token_from"

(S105)